app/models/user.rb

```diff
@@ -4,21 +4,24 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  has_many :pending_purchases
+  has_many :line_items
+  has_one :cart
+
+  after_save :create_cart
+
+  def create_cart
+    Cart.create!(user: self)
+  end
 
   def add_to_cart(book)
```

Comment:
Oh interesting, I might be overlooking something, since I see that the user has `line_items` as well. Where is this relationship used?